.github/workflows/test.yml

```diff
@@ -24,8 +24,19 @@ jobs:
           restore-keys: npm-v2-
       - run: npm ci
         if: steps.cache.outputs.cache-hit != 'true'
-      - run: npm run lint
-      - run: npm run test
+      - name: lint
+        run: npm run lint
+      - name: test
+        run: npm run test
+      - name: eth-gas-report
+        run: npm run test
+        env:
+          REPORT_GAS: 1
+          CC_SECRET: ${{ secrets.CC_SECRET }}
```

Comment:
I think you'll also need to run an `npx codechecks` command separately after the tests complete. This retrieves historical data from the codechecks backend to construct the comparison diff and triggers the check in GH actions.

@Amxx Were you able to get a CC_SECRET in the private beta at codechecks?

cf: [codechecks docs at eth-gas-reporter][1]

[1]: https://github.com/cgewecke/eth-gas-reporter/blob/master/docs/codechecks.md#setup

Reply:
thx for the help. I'm trying to get that working and don't really understand everything yet

Reply:
Running `npx codechecks` gives:
```Error occured:  Provider should never be in fork mode and not in PR mode!```

Do you have any idea ho to fix that ?

Reply:
Mmm. Not sure. The codechecks GH Actions integration might not work well with PRs from forks...

Will investigate.

Reply:
@Amxx Not certain but this error might be a quirk caused by adding codechecks to the PR after the PR has already been opened. There's a comment in the GH Actions code at [codechecks/monorepo][1] which says:

> // github actions have explicit event for PR creation but we cant use it since it fired only once
> // in future we need to build a mechanism to get pull request information using API since it's impossible to get from the inside the PR

You could test this hypothesis by re-opening as different PR? Or merging into a staging branch? 


[1]: https://github.com/codechecks/monorepo/blob/ba52ad88a50898bf4ad246fd16ce5f7f7ada4034/packages/client/src/ci-providers/GithubActions.ts#L4-L5

Reply:
We are going to merge a version that generate report without CI integration for now. Will open an issue to do that in a following PR.